File: src/utils/environment.js
/**
 * @param {import("../api/environment.js").Environment} environment
 */
export function mapEnvironmentFromApi(environment) {
  return {
    id: environment.id,
    project_id: environment.project_id,
    key: String(environment.id),
    name: environment.name,
    branch: environment.git_branch,
    url: environment.url,
    description: environment.description,
    git_branch: environment.git_branch,
    ssh_host: environment.ssh_host,
    ssh_password: environment.ssh_password,
    db_host: environment.db_host,
    db_port: environment.db_port,
    db_name: environment.db_name,
    db_username: environment.db_username,
    db_password: environment.db_password,
    created_at: environment.created_at,
    updated_at: environment.updated_at,
  };
}

File: src-tauri/src/models/environment.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Environment {
    pub id: i64,
    pub project_id: i64,
    pub name: String,
    pub url: Option<String>,
    pub git_branch: Option<String>,
    pub ssh_host: Option<String>,
    pub ssh_password: Option<String>,
    pub db_host: Option<String>,
    pub db_port: Option<i32>,
    pub db_name: Option<String>,
    pub db_username: Option<String>,
    pub db_password: Option<String>,
    pub description: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Deserialize)]
pub struct CreateEnvironment {
    pub project_id: i64,
    pub name: String,
    pub url: Option<String>,
    pub git_branch: Option<String>,
    pub ssh_host: Option<String>,
    pub ssh_password: Option<String>,
    pub db_host: Option<String>,
    pub db_port: Option<i32>,
    pub db_name: Option<String>,
    pub db_username: Option<String>,
    pub db_password: Option<String>,
    pub description: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
pub struct UpdateEnvironment {
    pub id: i64,
    pub name: String,
    pub url: Option<String>,
    pub git_branch: Option<String>,
    pub ssh_host: Option<String>,
    pub ssh_password: Option<String>,
    pub db_host: Option<String>,
    pub db_port: Option<i32>,
    pub db_name: Option<String>,
    pub db_username: Option<String>,
    pub db_password: Option<String>,
    pub description: Option<String>,
}
